File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.java
class Solution {
    public boolean increasingTriplet(int[] nums) {
//         initialize two integers to the maximum value
        int pointer1 = Integer.MAX_VALUE, pointer2 = Integer.MAX_VALUE;
        
        for (int n : nums) {
            if (n <= pointer1) {
                pointer1 = n;
            } else if (n <= pointer2) {
                pointer2 = n;
            } else {
                return true;
            }
        }
        return false;
        
    }
}

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var increasingTriplet = function(nums) {
    var num1 = Number.MAX_VALUE;
    var num2 = Number.MAX_VALUE;
    
    for(let i = 0; i < nums.length; i++) {
        if (nums[i] <= num1) {
            num1 = nums[i];
        } else if (nums[i] <= num2) {
            num2 = nums[i];
        } else {
            return true;
        }
    };
    
    return false;
};